executor/adapter.go

```diff
@@ -437,26 +445,72 @@ func (a *ExecStmt) LogSlowQuery(txnTS uint64, succ bool) {
 	}
 }
 
-func (a *ExecStmt) getStatsInfo() map[string]uint64 {
-	var physicalPlan plannercore.PhysicalPlan
-	switch p := a.Plan.(type) {
-	case *plannercore.Insert:
-		physicalPlan = p.SelectPlan
-	case *plannercore.Update:
-		physicalPlan = p.SelectPlan
-	case *plannercore.Delete:
-		physicalPlan = p.SelectPlan
-	case plannercore.PhysicalPlan:
-		physicalPlan = p
+// LogExpensiveQuery logs the queries which exceed the time threshold or memory threshold.
+func LogExpensiveQuery(ctx context.Context, sctx sessionctx.Context, ticker *time.Ticker, p plannercore.Plan, sql string) {
+	level := log.GetLevel()
+	if level > zapcore.WarnLevel {
+		return
+	}
+	if ticker != nil {
+		defer ticker.Stop()
+		<-ticker.C
```

Comment:
If the ticker is already closed, will it still log the query?

Reply:
No, https://play.golang.org/p/Rgc_UDvHv6a

Reply:
https://play.golang.org/p/R7kMIiy5lyc, seems that this function will never quit in this case.